src/harp-protocol/harp/protocol/base.py

```diff
@@ -0,0 +1,280 @@
+from datetime import datetime
+from enum import IntEnum, IntFlag
+
+# The reference epoch for UTC harp time
+REFERENCE_EPOCH = datetime(1904, 1, 1)
+
+# Bit masks for the PayloadType
+_isUnsigned: int = 0x00
+_isSigned: int = 0x80
+_isFloat: int = 0x40
+_hasTimestamp: int = 0x10
+
+
+class MessageType(IntEnum):
+    """
+    An enumeration of the allowed message types of a Harp message. More information on the MessageType byte of a Harp message can be found [here](https://harp-tech.org/protocol/BinaryProtocol-8bit.html#messagetype-1-byte).
+
+    Attributes
+    ----------
+    READ : int
+        The value that corresponds to a Read Harp message (1)
+    WRITE : int
+        The value that corresponds to a Write Harp message (2)
+    EVENT : int
+        The value that corresponds to an Event Harp message (3). Messages of this type are only meant to be send by the device
+    READ_ERROR : int
+        The value that corresponds to a Read Error Harp message (9). Messages of this type are only meant to be send by the device
+    WRITE_ERROR : int
+        The value that corresponds to a Write Error Harp message (10). Messages of this type are only meant to be send by the device
+    """
+
+    READ = 1
+    WRITE = 2
+    EVENT = 3
+    READ_ERROR = 9
+    WRITE_ERROR = 10
+
```

Comment:
```suggestion
class MessageType(IntFlag):
    READ = 0x01
    WRITE = 0x02
    EVENT = 0x03
    ERROR = 0x08
    READ_ERROR = READ | ERROR
    WRITE_ERROR = WRITE | ERROR
```

Consider using a IntFlag instead. 
It is kinda weird that:
print(Foo.READ & Foo.EVENT) evaluates to foo.READ, but it is also the case using the current Enum anyway.

Reply:
We will look into your suggestion.